### DISCUSSION TITLE
Post request does not work if one parameter is provided in function.

### DISCUSSION THREAD

**Discussion by daviddanielng** [fastapi/fastapi#12456] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [x] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

from fastapi import Body, FastAPI

app = FastAPI()


@app.post("/test")
async def test(name: Annotated[str, Body(..., alias="name")]):
    # this will raise a validation error and return a 422 response
    return {"name": name}


@app.post("/test2")
async def test2(
    name: Annotated[str, Body(..., alias="name")],
    short_name: Annotated[str, Body(..., alias="short_name")],
):
    #  this works fine
    return {"name": name, "short_name": short_name}


@app.post("/test3")
async def test3(
    name: Annotated[str, Body(..., alias="name")],
    short_name: Annotated[str, Body(..., alias="name")],
):
    #  this also works fine
    return {"name": name, "short_name": short_name}
```

### Description

Call endpoint `/test` with body `{"name":"hello"}` or `{"name":"hello","short_name":"h"}`
I am expecting `{"name": "hello"}` I got  
```json
{
    "detail": [
        {
            "type": "string_type",
            "loc": [
                "body"
            ],
            "msg": "Input should be a valid string",
            "input": {
                "name": "hello",
                "short_name": "h"
            }
        }
    ]
}
```
Now call `/test2`  with body `{"name":"hello","short_name":"h"}` this return exactly what i expected 
```json
{
    "name": "hello",
    "short_name": "h"
}
```

Now call `/test3` with body `{"name":"hello"}` this return the same thing as `/test2` with change to `short_name`
```json
{
    "name": "hello",
    "short_name": "hello"
}
```

Now, I am really confused, I looked at the swagger docs and saw that, `test` is expecting a body of string while `test2` and `test3` is expecting `json`, is this the intended behaviour?

### Operating System

Linux

### Operating System Details

NAME="Ubuntu"
VERSION="20.04.6 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.6 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal

### FastAPI Version

0.115.2

### Pydantic Version

2.9.2

### Python Version

3.12.1

### Additional Context

If I am doing it wrong, please let me know.
I want fastapi to do validation for me,  I am already handling errors with
```python
@app.exception_handler(RequestValidationError)
async def validation_exception_handler(
    request: Request, exc: RequestValidationError
) -> JSONResponse:
    message = convert_fast_api_error_to_message(exc)

    return JSONResponse(
        status_code=StatusCodes.UNPROCESSABLE_ENTITY.value,
        content=APIResponse(data=message).model_dump(),
    )

```

**Comment by YuriiMotov** [comment, marked as answer]
When you have only one body parameter, FastAPI expects its value directly by default (`"hello"` instead of `{"name": "hello"}`).

To change this behavior you can set the `embed` parameter to `True`:
```
@app.post("/test")
async def test(name: Annotated[str, Body(..., alias="name", embed=True)]):
    # this will work as you expected
    return {"name": name}
```

You can read more here: https://fastapi.tiangolo.com/tutorial/body-multiple-params/?h=embed#multiple-body-params-and-query

**Reply by daviddanielng** [reply]
Thanks.
